src/components/date-input/DateInput.tsx

```diff
@@ -198,17 +214,22 @@ class DateInput extends PureComponent<DateInputProps, DateInputState> {
     };
 
     return (
-      <div className={classNames('nhsuk-date-input', className)} {...rest} onChange={this.onChange}>
+      <>
+      {this.props.label ? <Label htmlFor={labelHtmlFor}>{this.props.label}</Label> : ''}
+      {this.props.hint ? <Hint>{this.props.hint}</Hint> : ''}
+      {this.props.error ? <ErrorMessage>{this.props.error}</ErrorMessage> : ''}
+      <div id = {this.props.id} className={classNames('nhsuk-date-input', className)} {...rest} onChange={this.onChange}>
         <DateInputContext.Provider value={contextValue}>
           {children || (
             <>
-              <DateInput.Day />
-              <DateInput.Month />
-              <DateInput.Year />
+              <DateInput.Day error={this.props.error} />
+              <DateInput.Month error={this.props.error}/>
+              <DateInput.Year error={this.props.error} />
```

Comment:
As the error is passed in to each component here, could this be passed to the components using context instead?

Reply:
Good point! Very new to React/Front End in general, so hadn't heard of it. But I've given adding to the existing context a go, is the change around this what you meant?

Reply:
Haven't really touched context before (I'm quite new to react/frontend stuff in general) but I gave it a go, is this what you meant?